apps/omg_watcher/lib/omg_watcher/monitor.ex

```diff
@@ -0,0 +1,157 @@
+# Copyright 2018 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.Watcher.Monitor do
```

Comment:
how DRYable this is wrt `OMG.API.Monitor` - they look very similar. Maybe it belongs to `omg` instead?

Reply:
That would mean passing children towards omg, and not fetching them from it. It would change the behaviour of who belongs to whom. I think we should keep the implied semantics `watcher has omg`, or `childchain has omg`.

Reply:
hm, maybe at least some common part which doesn't include the children could be refactored out. But let's tackle this elsewhere, I'd need to get more into detail here

Reply:
It's really a balance between being dry and keeping the process state contained in a module in this case. I prefer readability over dryness tbh.